openjdk/mmtkBarrierSet.hpp

```diff
@@ -87,57 +116,44 @@ class MMTkBarrierSet : public BarrierSet {
   // 1) Provide an enum "name" for the BarrierSet in barrierSetConfig.hpp
   // 2) Make sure the barrier set headers are included from barrierSetConfig.inline.hpp
   // 3) Provide specializations for BarrierSet::GetName and BarrierSet::GetType.
+
+  #define UNREACHABLE() guarantee(false, "UNREACHABLE");
   template <DecoratorSet decorators, typename BarrierSetT = MMTkBarrierSet>
   class AccessBarrier: public BarrierSet::AccessBarrier<decorators, BarrierSetT> {
   private:
     typedef BarrierSet::AccessBarrier<decorators, BarrierSetT> Raw;
   public:
     template <typename T>
     static void oop_store_in_heap(T* addr, oop value) {
-      Raw::oop_store(addr, value);
-      if (MMTkBarrierSet::enable_write_barrier) {
-        MMTkBarrierRuntime::record_modified_edge((void*) addr);
-      }
+      UNREACHABLE();
```

Comment:
What's the reason that some of the methods are unreachable now?

Reply:
Because these methods are never called by hotspot (at lease when running dacapo benchmarks). Probably they are just legacy interfaces...